update/v1alpha1/types_clusteroperator_progress_insight.go

```diff
@@ -0,0 +1,133 @@
+	// - Updating: whether the operator is updating; When Updating=False, the reason field can be Pending or Updated
+	// - Healthy: whether the operator is considered healthy; When Healthy=False, the reason field can be Unavailable or Degraded, and Unavailable is "stronger" than Degraded
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// name is the name of the operator
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
+	Name string `json:"name"`
+
+	// resource is the ClusterOperator resource that represents the operator
+	//
+	// +Note: By OpenShift API conventions, in isolation this should be a specialized reference that refers just to
+	// +resource name (because the rest is implied by status insight type). However, because we use resource references in
+	// +many places and this API is intended to be consumed by clients, not produced, consistency seems to be more valuable
+	// +than type safety for producers.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.group == 'config.openshift.io' && self.resource == 'clusteroperators'",message="resource must be a clusteroperators.config.openshift.io resource"
+	Resource ResourceRef `json:"resource"`
+}
+
+// ClusterOperatorProgressInsightConditionType are types of conditions that can be reported on ClusterOperator status insights
+type ClusterOperatorProgressInsightConditionType string
+
+const (
+	// Updating condition communicates whether the ClusterOperator is updating
+	ClusterOperatorProgressInsightUpdating ClusterOperatorProgressInsightConditionType = "Updating"
+	// Healthy condition communicates whether the ClusterOperator is considered healthy
+	ClusterOperatorProgressInsightHealthy ClusterOperatorProgressInsightConditionType = "Healthy"
+)
+
+// ClusterOperatorUpdatingReason are well-known reasons for the Updating condition on ClusterOperator status insights
+type ClusterOperatorUpdatingReason string
+
+const (
+	// Updated is used with Updating=False when the ClusterOperator finished updating
+	ClusterOperatorUpdatingReasonUpdated ClusterOperatorUpdatingReason = "Updated"
+	// Pending is used with Updating=False when the ClusterOperator is not updating and is still running previous version
+	ClusterOperatorUpdatingReasonPending ClusterOperatorUpdatingReason = "Pending"
+	// Progressing is used with Updating=True when the ClusterOperator is updating
+	ClusterOperatorUpdatingReasonProgressing ClusterOperatorUpdatingReason = "Progressing"
+	// CannotDetermine is used with Updating=Unknown
+	ClusterOperatorUpdatingCannotDetermine ClusterOperatorUpdatingReason = "CannotDetermine"
```

Comment:
nit: there doesn't seem to be any enum-style validation on `conditions[].reason`, so maybe we can leave these off and go free-form in the resource-creating controller?  We'd need `Updated` or `Pending` for the `Updating=False` case to allow clients to distinguish there, but I'm not clear on why clients would need to distinguish by well-known reason for `Updating=True` or `Updating=Unknown`, especially with us only declaring one `reason` string for each of those `status` values.  But also, obviously not a big deal if we wanted central definitions of some default reasons, in case that helps make resource-creators more consistent in the "there's no more-granular information we want to share" cases.

Reply:
I'd more inclined to add validations eventually rather than drop to free-form, because I think we should either be full "reasons are well-known and can matter" or full  "reasons are free-form", but not half-way. I'll add the CEL rules.

Reply:
I have started adding validations for conditions (to ensure condition presence and that they use the correct reasons etc) but then I realized tightening the API this much at this point makes it harder to experiment in the controller (for example, I'm not sure if I can get away with a single `Healthy` condition on CV & MCP insights, or whether I would need to break it down. For now the API targets DevPreview and does not need to be this tight, so for now I only added TODOs everywhere.